src/node/iir_filter.rs

```diff
@@ -116,96 +198,66 @@ impl IIRFilterNode {
         &self,
-        frequency_hz: &mut [f32],
+        frequency_hz: &[f32],
         mag_response: &mut [f32],
         phase_response: &mut [f32],
     ) {
-        self.validate_inputs(frequency_hz, mag_response, phase_response);
+        if frequency_hz.len() != mag_response.len() || mag_response.len() != phase_response.len() {
+            panic!("InvalidAccessError - Parameter lengths must match");
+        }
+
         let sample_rate = self.context().sample_rate() as f64;
+        let nquist = sample_rate / 2.;
 
         for (i, &f) in frequency_hz.iter().enumerate() {
+            let freq = f64::from(f).clamp(0., nquist);
+            let z = -2.0 * PI * freq / sample_rate;
             let mut num: Complex<f64> = Complex::new(0., 0.);
             let mut denom: Complex<f64> = Complex::new(0., 0.);
 
-            // 0 through 20 casts without loss of precision
-            #[allow(clippy::cast_precision_loss)]
-            for (idx, &ff) in self.feedforward.iter().enumerate() {
-                num += Complex::from_polar(ff, idx as f64 * -2.0 * PI * f64::from(f) / sample_rate);
+            for (idx, &b) in self.feedforward.iter().enumerate() {
+                num += Complex::from_polar(b, idx as f64 * z);
             }
 
-            // 0 through 20 casts without loss of precision
-            #[allow(clippy::cast_precision_loss)]
-            for (idx, &fb) in self.feedback.iter().enumerate() {
-                denom +=
-                    Complex::from_polar(fb, idx as f64 * -2.0 * PI * f64::from(f) / sample_rate);
+            for (idx, &a) in self.feedback.iter().enumerate() {
+                denom += Complex::from_polar(a, idx as f64 * z);
             }
 
-            let h_f = num / denom;
+            let response = num / denom;
 
-            // Possible truncation is fine. f32 precision should be sufficients
-            // And it is required by the specs
-            mag_response[i] = h_f.norm() as f32;
-            phase_response[i] = h_f.arg() as f32;
-        }
-    }
-
-    /// validates that the params given to `get_frequency_response` method
-    #[inline]
-    fn validate_inputs(
-        &self,
-        frequency_hz: &mut [f32],
-        mag_response: &mut [f32],
-        phase_response: &mut [f32],
-    ) {
-        assert_eq!(
-            frequency_hz.len(),
-            mag_response.len(),
-            " InvalidAccessError: All paramaters should be the same length"
-        );
-        assert_eq!(
-            mag_response.len(),
-            phase_response.len(),
-            " InvalidAccessError: All paramaters should be the same length"
-        );
-
-        // Ensures that given frequencies are in the correct range
-        let min = 0.;
-        let max = self.context().sample_rate() / 2.;
-        for f in frequency_hz.iter_mut() {
-            *f = f.clamp(min, max);
+            let (mag, phase) = response.to_polar();
+            mag_response[i] = mag as f32;
```

Comment:
For performance (not really important here) you could zip them along in the loop, this prevents bounds checks on every loop item:
```
for (i, &f, m, p) in frequency_hz.iter().enumerate().zip(mag_response.iter_mut()).zip(phase_response.iter_mut())
```